Add feature: archive tasks regulary

i-am-harveyt opened the pull request:
1. The tasks are all automatically archived(00:00 every Sunday)
2. 正在玩 正在玩 LeetCode -> 正在玩 LeetCode

Commit:
Add feature: archive tasks regulary
1. The tasks are all automatically archived(00:00 every Sunday)
2. 正在玩 正在玩 LeetCode -> 正在玩 LeetCode

Commit:
Suppliment: `!task-archive` is useless now

shinyu0430 commented on utils/archive.js:
```diff
@@ -0,0 +1,48 @@
+    // archive data
+    let archiveData = [];
+    const date = new Date();
+    for (let i = 0; i < data.length; i++) {
+      // Increase the seconds of the date by the index value
+      date.setSeconds(date.getSeconds() + i);
+
+      // Convert the date to a string formatted as YYYY-MM-DD HH:MI:SS
+      const formattedDate = `${date
+        .toISOString()
+        .replace("T", " ")
+        .replace(/\.\d+Z$/, "")}`;
+
+      archiveData.push({
+        guild_id: data[i]["guild_id"],
+        user_id: data[i]["user_id"],
+        task_name: data[i]["task_name"],
+        archived_at: formattedDate,
+      });
+    }
```
I think it would be more concise to use map instead of a for loop.
```javascript
const archiveData = data.map((task, index) => {
  const formattedDate = new Date(date.getTime() + index * 1000)
    .toISOString()
    .replace("T", " ")
    .replace(/\.\d+Z$/, "");

  return {
    guild_id: task["guild_id"],
    user_id: task["user_id"],
    task_name: task["task_name"],
    archived_at: formattedDate,
  };
});
```

i-am-harveyt replied:
Sure, I'll use this instead.

shinyu0430 commented on utils/archive.js:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    // remove old data
+    await client.from("tasks").delete().in("done", [true, false]);
```
Adding error handling would be better.

i-am-harveyt replied:
OK

shinyu0430 commented on utils/archive.js:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    // insert to archived table
+    await client.from("archived").insert(archiveData);
```
I think it would be better to archive the data first instead of deleting it first. In this way, we can ensure not accidentally lose users' data when the process of archiving goes wrong.

i-am-harveyt replied:
Sure, it seems like a better choice.

shinyu0430 commented on utils/archiveRoutine.js:
```diff
@@ -0,0 +1,18 @@
+
+// cron, do the archive task
+const archive_tasks = new cron.CronJob(
+	"0 0 0 * * 0", // do each 30s
```
remove the wrong-meaning comment.

i-am-harveyt replied:
Sorry, my fault. I'll write a accurate one.

shinyu0430 commented:
I apologize for the delayed code review, and I sincerely appreciate your effort. Please kindly consider the review feedback I have provided.

i-am-harveyt committed:
Fix: meet the reviewer's request
1. Error handling in `archive.js`
2. Delete after insertion in `archive.js`
1. Re-write the comment in `archiveRoutine.js`

shinyu0430 commented:
Thank you!
